src/java.sql/share/classes/java/sql/Statement.java

```diff
@@ -779,7 +779,7 @@
      *
      * @throws SQLException if a database access error occurs,
      * this method is called on a closed {@code Statement}, the SQL
-     * statement returns a {@code ResultSet} object,the second argument
+     * statement returns a {@code ResultSet} object, the second argument
      * supplied to this method is not an
      * {@code int} array whose elements are valid column indexes, the method is called on a
```

Comment:
Should it be "or the method is called on...", i.e. add the "or", otherwise it's a list of problems but we don't know if they are all required, or are alternatives.  It probably does not mean that all these have to be true to throw the exception, but it doesn't say that.  We are nitpicking of course.

Reply:
You are right in that this `@throws` description reads a bit weird in its current form. That said, I wouldn't touch it in this PR for two reasons. Firstly, this wording seems to be consistent with other locations in this file. Secondly, this is a spec territory.

Reply:
Yes an "or" is probably worthwhile to add.

Reply:
> Yes an "or" is probably worthwhile to add.

I would prefer to leave it for the follow-up cleanup if only because adding "or" here would make it inconsistent with other `@throws SQLException` descriptions in that file and perhaps elsewhere in java.sql:

* java/sql/Statement.java:59
* java/sql/Statement.java:85
* java/sql/Statement.java:346
* java/sql/Statement.java:524
* java/sql/Statement.java:745
* java/sql/Statement.java:814
* java/sql/Statement.java:860
* java/sql/Statement.java:913
* java/sql/Statement.java:962
* java/sql/Statement.java:1225
* java/sql/Statement.java:1269
* java/sql/Statement.java:1314

Reply:
> > Yes an "or" is probably worthwhile to add.
> 
> I would prefer to leave it for the follow-up cleanup if only because adding "or" here would make it inconsistent with other `@throws SQLException` descriptions in that file and perhaps elsewhere in java.sql:
> 
> * java/sql/Statement.java:59
> * java/sql/Statement.java:85
> * java/sql/Statement.java:346
> * java/sql/Statement.java:524
> * java/sql/Statement.java:745
> * java/sql/Statement.java:814
> * java/sql/Statement.java:860
> * java/sql/Statement.java:913
> * java/sql/Statement.java:962
> * java/sql/Statement.java:1225
> * java/sql/Statement.java:1269
> * java/sql/Statement.java:1314

I am fine with that.  I guess a semi-colon could also be used vs. a comma to divide the Exception scenarios listed